# Genome size scripts
Leptidea genome size scripts

# jellyfish.sh
This is the script bares the code used to generate individual histogram files for each sample. This script uses the fastq files provided in the project repository. This script was run on 16 cores using K-mer length as 17. The final output of this script gives a histogram file(.hist) for each sample which will then be used for genome size estimates using the R script “plots_generator.R”
Dependencies: Jellyfish software. 

# reads.txt
This file contains number of reads subsampled to obtain approximately 5X coverage for each sample. 
Format: Sample_ID     Number of reads     Average read length

# test.hist
This file contains a concatenated file of the K-Mer histograms generated by jellyfish.sh. This file will be used by the R script “plots_generator.R” to estimate genome size of each sample. 
Format: Multiplicity	K-mer counts 		Sample_ID 		Population


# plots_generator.R
This is the R script that contains the code that estimates the genome size from the K-mer histogram. It uses the above mentioned files within its pipeline. 
Dependencies: R libraries ggplot2, ggfortify, cluster, grid


# genome_size_newdata_trimmed_K17_subsample_1.zip
Raw outputs and preprocessing .histo files for each sample with R script and unscaled genome sizes. 
